### PR TITLE
Add a baseline e2e pipeline

### LINKED ISSUE #1: Set up a project
This issue is to set up a project and move some basic work that was already done offline.
- [x] create a project structure (main file and separate ones for utils)
- [x] move code from old repo 
- [x] implement functions for:
    - [x] data loading 
    - [x] model evaluation
    - [x] extracting submission in correct format
- [x] train a baseline model (simple - that will work as a reference point)

### PR DESCRIPTION
**Pull request by jakubkarczewski**
This is a baseline e2e pipeline containing:
- tabular data loading
- basic data cleansing
- simple LGBM model with KFold CV
- rounding LGBM outputs via OptimizedRounder (links in code - it's cool)
- submission creation

Feel free to comment and criticize :p

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    # print distributions of predictions vs. true distributions
+    print("\nTrue Distribution:")
+    print(pd.value_counts(y_train, normalize=True).sort_index())
+    print("\nTrain Predicted Distribution:")
+    print(pd.value_counts(rounded_train_outputs, normalize=True).sort_index())
+    print("\nTest Predicted Distribution:")
+    print(pd.value_counts(rounded_test_outputs, normalize=True).sort_index())
+
+    # generate submission
```
Comment (Adrianna-Na) [inline main.py:54]: Write a function to generate submissions and add to data_utils. Also maybe it makes sense to add parameter for output dir here?

File: models.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""This module contains model related code"""
+
+from collections import Counter
+
+import lightgbm as lgb
+from sklearn.model_selection import StratifiedKFold
+import numpy as np
+
+SEED = 997
```
Comment (Adrianna-Na) [inline models.py]: Why are we setting seed as a set value?

File: main.py (hunk 1 above)
Comment (jakubkarczewski) [reply main.py:54]: Ook, will do :)

File: models.py (hunk 2 above)
Comment (jakubkarczewski) [reply models.py]: If you mean "set value" as "constant" then it's for reproducibility sake. If you mean something else please clarify :)

File: config/path_config.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+"""Contains dataset's directory path"""
+DATA_DIR = '/home/kuba/Development/Datasets/PetFinder'
```
Comment (Adrianna-Na) [inline config/path_config.py]: We agreed not to push this to master branch.
fyi: https://stackoverflow.com/questions/4308610/how-to-ignore-certain-files-in-git/44132149

**Review by Adrianna-Na — CHANGES_REQUESTED** [review]

File: config/path_config.py (hunk 3 above)
Comment (jakubkarczewski) [reply config/path_config.py]: fck sry

**Review by Adrianna-Na — APPROVED** [review]

**Review by MichalGawryluk — APPROVED** [review]
Fine. Especially after Ada's comments.
